### PR TITLE
Split `toc.rs` into sub-modules

### PR DESCRIPTION
**Pull request by ffuugoo**
Refactor `toc.rs` and `toc_components/tmp_directories.rs` into:

```
toc/
└── mod.rs
└── collection_container.rs
└── collection_meta_ops.rs
└── create_collection.rs
└── locks.rs
└── point_ops.rs
└── snapshots.rs
└── temp_directories.rs (former toc_components/tmp_directories.rs)
```

Most files are under 300 LOC now. `toc/mod.rs` is 600 lines, but I think it's still acceptable.

I've organized commit history in a way that is easy to track **commit-by-commit**:

- each commit touch at most two files (`toc/mod.rs` and one of the submodules)
- code always moved in blocks, so that it's relatively easy to compare moved blocks side-by-side
- I didn't make any changes to the moved code...
  - except I've changed a few calls from `function(...)` to `mod::function(...)`
  - but these changes are only made in `Add explicit imports to ...` commits, and should be easy to track

Methods in each submodule reorganized by (roughly) the following principles:

- public before private
- general before implementation details
  - e.g., if method `A` calls method `B`, I'd put method `A` before method `B`
- everything is kept close to the point of first use
- and sorted in the order of usage

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [x] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?

### CODE REVIEW

File: lib/storage/src/content_manager/toc/mod.rs [hunk 1]
```diff
@@ -782,84 +585,6 @@ impl TableOfContent {
...
-        //  │ Shard: Some(N)
-        //  │ Ordering: Strong
-        // ┌▼──────────────────┐
-        // │ Leader node       │ <- update_from_peer
-        // └┬──────────────────┘
-        //  │ Shard: Some(N)
-        //  │ Ordering: None(Weak)
-        // ┌▼──────────────────┐
-        // │ Updating node     │ <- update_from_peer
-        // └───────────────────┘
```
Comment (timvisee) [inline lib/storage/src/content_manager/toc/mod.rs]: I'm missing this comment after refactoring (7c5f00a70307a0c9ee646f107ccfffb9c05b0b66). I think it'll be useful to keep.

**Review by timvisee — COMMENTED** [review]
Thanks a **lot** for packing these changes into separate commits. Helps with going through 2k changes :)

File: lib/storage/src/content_manager/toc/mod.rs (hunk 1 above)
Comment (ffuugoo) [reply lib/storage/src/content_manager/toc/mod.rs]: Nice catch. I'll add it back now.

Comment (ffuugoo) [reply lib/storage/src/content_manager/toc/mod.rs]: Done.

**Review by generall — APPROVED** [review]

**Comment by generall** [comment]
reviewing & merging this fast, to avoid conflicts

**Pull request merged**